City-of-Helsinki/helsinki-paatokset
# PP-228: Handling cancelled meetings

**To test**
- Checkout branch, run `make drush-cr drush-cim`
- Make sure you have some meetings and ahjo content by running `make ahjo-migrations`

**Cancelled meetings migration**
- Run `drush migrate-status ahjo_meetings:cancelled` to check that the migration exists and content has been aggregated in the test environment
- Then run `drush migrate-import ahjo_meetings:cancelled --update`
- Check the content list for meetings: https://helsinki-paatokset.docker.so/fi/admin/content/meetings 
  - Some of those should have the status "peruttu" 

**Cancelled meetings in policymaker pages**
- Find some meetings for any policymaker and edit them:
  - An upcoming meeting
  - A meeting with agenda or minutes published
- Check that these are visible in the policymaker calendar and in the documents list
- Edit the meetings and change the "Status" field to "peruttu"
  - The cancelled meeting should appear with it's title and date stricken through and with a "Meeting cancelled" text
  - The cancelled meeting should disappear from the documents list
- Check that the "Meeting cancelled" text is translatable through the UI translation 

**Cancelled meetings in the frontpage calendar**
- Find a meeting or meetings that appear in the frontpage calendar
- Edit them and change the "Status" field to "peruttu"
- They should disappear from the frontpage

**Cancelled meetings in the JS calendar**
- Open the meetings calendar: https://helsinki-paatokset.docker.so/fi/kokouskalenteri
- Find the upcoming meetings you've edited and cancelled. These should appear without any meeting links and with their titles and datest stricken through and with a "meeting cancelled" text.
- Edit a meeting and change it's meeting date so it's different from the original date
- Edit some past meetings and change their status to "peruttu"
- The moved dates should appear in the calendar with a "meeting moved" text.
- The cancelled meetings in the past should disappear from the calendar
- Check that the "meeting cancelled" and "meeting moved" texts are translatable through the UI translation
